iscservice/demo/util.c

```diff
@@ -0,0 +1,710 @@
+	fclose(fp);
+	if (rc < 0) {
+		ERR("failed to read parameters from %s: rc=%d", c0rcfile, rc);
+		return rc;
+	}
+	rc_params_nr = rc;
+
+	rc = pools_fids_set(rc_params, rc_params_nr);
+	if (rc <= 0) {
+		if (rc == 0) {
+			ERR("no pools configured at %s\n", c0rcfile);
+		} else {
+			ERR("failed to set pools from %s\n", c0rcfile);
+		}
+		return -EINVAL;
+	}
+
+	rc = read_conf_params(idx, rc_params, rc_params_nr);
+	if (rc != 0) {
+		ERR("failed to read conf parameters from %s\n", c0rcfile);
+		return rc;
+	}
+
+	m0_conf.mc_is_oostore            = true;
+	m0_conf.mc_is_read_verify        = false;
+#if 0
+	/* set to default values */
+	m0_conf.mc_tm_recv_queue_min_len = M0_NET_TM_RECV_QUEUE_DEF_LEN;
+	m0_conf.mc_max_rpc_msg_size      = M0_RPC_DEF_MAX_RPC_MSG_SIZE;
+#endif
+	/* set to Sage cluster specific values */
+	m0_conf.mc_tm_recv_queue_min_len = 64;
+	m0_conf.mc_max_rpc_msg_size      = 65536;
+	m0_conf.mc_layout_id             = 0;
+	if (unit_size) {
+		lid = m0_obj_unit_size_to_layout_id(unit_size * 1024);
+		if (lid == 0) {
+			fprintf(stderr, "invalid unit size %u, it should be: "
+				"power of 2, >= 4 and <= 4096\n", unit_size);
+			return -EINVAL;
+		}
+		m0_conf.mc_layout_id = lid;
+	}
+
+	/* IDX_MOTR */
+	m0_conf.mc_idx_service_id   = M0_IDX_DIX;
+	dix_conf.kc_create_meta = false;
+	m0_conf.mc_idx_service_conf = &dix_conf;
+
+#if DEBUG
+	fprintf(stderr,"\n---\n");
+	fprintf(stderr,"%s,", (char *)m0_conf.mc_local_addr);
+	fprintf(stderr,"%s,", (char *)m0_conf.mc_ha_addr);
+	fprintf(stderr,"%s,", (char *)m0_conf.mc_profile);
+	fprintf(stderr,"%s,", (char *)m0_conf.mc_process_fid);
+	fprintf(stderr,"%s,", (char *)m0_conf.mc_idx_service_conf);
+	fprintf(stderr,"\n---\n");
+#endif
+
+	if (!m0trace_on)
+		m0_trace_set_mmapped_buffer(false);
+	/* m0_instance */
+	rc = m0_client_init(&m0_instance, &m0_conf, true);
+	if (rc != 0) {
+		fprintf(stderr, "failed to initilise the Client API\n");
+		return rc;
+	}
+
+	/* And finally, client root realm */
+	m0_container_init(&container, NULL, &M0_UBER_REALM,
+				 m0_instance);
+	rc = container.co_realm.re_entity.en_sm.sm_rc;
+	if (rc != 0) {
+		fprintf(stderr,"failed to open uber realm\n");
+		return rc;
+	}
+
+	m0_list_init(&isc_reqs);
+
+	/* success */
+	uber_realm = container.co_realm;
+	return 0;
+}
+
+/*
+ * m0util_free()
+ * free client resources.
+ */
+int m0util_free(void)
+{
+	m0_client_fini(m0_instance, true);
+	memset(c0rcfile, 0, sizeof(c0rcfile));
+	return 0;
+}
+
+/*
+ * m0util_setrc()
+ * set c0apps resource filename
+ */
+int m0util_setrc(char *prog)
+{
+	char hostname[MAX_RCFILE_NAME_LEN / 2] = {0};
+
+	/* null */
+	if (!prog) {
+		fprintf(stderr, "error! null progname.\n");
+		return -EINVAL;
+	}
+
+	gethostname(hostname, MAX_RCFILE_NAME_LEN / 2);
+	snprintf(c0rcfile, MAX_RCFILE_NAME_LEN, "%s/.m0util/%src/%s",
+		 getenv("HOME"), prog, hostname);
+
+	/* success */
+	return 0;
+}
+
+/*
+ * open_entity()
+ * open m0 entity.
+ */
+int open_entity(struct m0_entity *entity)
```

Comment:
This function is only used once.
Just move it to the location where it is used.

Reply:
Moved.